File: ios/Stax/Billing/SubscriptionState.swift
import Foundation

// MARK: – Subscription state

enum SubscriptionState {
    case free
    case premium(isInTrial: Bool, expiryDate: Date)
    case expired
}

// MARK: – Feature enum

enum Feature {
    case scan
    case aiScan
    case sessionCreate
    case photoAdd
    case chipConfig
    case favorites
    case shareClean
}

// MARK: – Limit check result

enum LimitResult {
    case allowed
    case softCap(message: String)
    case blocked(Feature)
}

// MARK: – Free tier constants

struct FreeTierLimits {
    static let maxScansPerDay        = 5
    static let maxSessions           = 3
    static let maxPhotosPerSession   = 10
    static let maxFavorites          = 3
    static let maxChipConfigCasinos  = 1
}

File: android/app/src/main/java/com/bitcraftapps/stax/data/billing/SubscriptionState.kt
package com.bitcraftapps.stax.data.billing

sealed interface SubscriptionState {
    data object Free : SubscriptionState
    data class Premium(val isInTrial: Boolean, val expiryMs: Long) : SubscriptionState
    data class Expired(val gracePeriodEndsMs: Long?) : SubscriptionState
}

enum class Feature {
    SCAN,
    AI_SCAN,
    SESSION_CREATE,
    PHOTO_ADD,
    CHIP_CONFIG,
    FAVORITES,
    SHARE_CLEAN
}

sealed interface LimitResult {
    data object Allowed : LimitResult
    data class SoftCap(val message: String) : LimitResult
    data class Blocked(val feature: Feature) : LimitResult
}

const val MAX_FREE_SCANS_PER_DAY = 5
const val MAX_FREE_SESSIONS = 3
const val MAX_FREE_PHOTOS_PER_SESSION = 10
const val MAX_FREE_FAVORITES = 3
const val MAX_FREE_CHIP_CONFIG_CASINOS = 1
